src/filesystem.cc

```diff
@@ -503,10 +513,19 @@ class GCSFileSystem : public FileSystem {
       google::cloud::StatusOr<gcs::ObjectMetadata>* metadata);
 
   google::cloud::StatusOr<gcs::Client> client_;
+  static std::mutex env_mu_;  // protect GOOGLE_APPLICATION_CREDENTIALS env var
 };
+std::mutex GCSFileSystem::env_mu_;
 
-GCSFileSystem::GCSFileSystem()
+GCSFileSystem::GCSFileSystem(const std::string& credential_path)
 {
+  static std::string env_name = "GOOGLE_APPLICATION_CREDENTIALS";
+  std::lock_guard<std::mutex> lock(env_mu_);
+#ifdef _WIN32
+  _putenv((env_name + "=" + credential_path).c_str());
+#else
+  setenv(env_name.c_str(), credential_path.c_str(), 1);
+#endif
   client_ = gcs::Client::CreateDefaultClient();
```

Comment:
There is no way to pass in the credential directly?

Reply:
Updated.

Reply:
Just found the way to pass credential via argument. Thanks for checking. Updated.